GitCommands/RevisionReader.cs

```diff
@@ -311,157 +312,163 @@ int CountParents(in ReadOnlySpan<byte> array, int baseOffset)
                 revision = default;
                 return false;
             }
 
+            offset += bytesConsumed + 1;
+
             #endregion
 
+            #region Encoded string values (names, emails, subject, body)
+
+            // The remaining must be decoded (for above utf8/ascii must work)
+            // First records are decoded on the stack
+            // The attributes are decoded in the order they are defined in the format string
             revision = new GitRevision(objectId)
             {
                 ParentIds = parentIds,
                 TreeGuid = treeId,
-                Author = author,
-                AuthorEmail = authorEmail,
+
+                Author = GetNextLine(buffer),
+                AuthorEmail = GetNextLine(buffer),
                 AuthorUnixTime = authorUnixTime,
-                Committer = committer,
-                CommitterEmail = committerEmail,
+                Committer = GetNextLine(buffer),
+                CommitterEmail = GetNextLine(buffer),
                 CommitUnixTime = commitUnixTime,
-                Subject = subject,
-                Body = body,
-                HasMultiLineMessage = hasMultiLineMessage,
-                HasNotes = false,
-                ReflogSelector = reflogSelector,
+                HasNotes = false
             };
 
-            return true;
+            // Body is occasionally big, like linux repo has 35K bytes, the buffer is over 100K
+            // Use a backing buffer on the heap
+            int maxChars = _logOutputEncoding.GetMaxByteCount(buffer.Slice(offset).Length);
+            if (maxChars > _decodeBuffer.Length)
+            {
+                // Default should be sufficient for most repos, Linux though has
+                // unencoded of 36K, which results in maxChars being greater than 100K
+                int newSize = _decodeBuffer.Length;
+                while (newSize < maxChars)
+                {
+                    newSize *= 2;
+                }
 
-            void ParseAssert(string? message)
+                _decodeBuffer = new char[newSize];
+            }
+
+            int decodedLength = _logOutputEncoding.GetChars(buffer.Slice(offset), _decodeBuffer);
+            offset = 0;
+            bool keepBody = authorUnixTime >= _oldestBody;
+
+            // reflogSelector are only used when listing stashes
+            if (_hasReflogSelector)
             {
-                _noOfParseError++;
-                Debug.Assert(_noOfParseError > 1, message);
-                Trace.WriteLineIf(_noOfParseError < 10, message);
+                revision.ReflogSelector = GetReflogSelector(_decodeBuffer);
             }
-        }
 
-        #region Nested type: StringLineReader
+            // Keep a full multiline message body within the last six months (by default).
+            // Note also that if body and subject are identical (single line), the body never need to be stored
```

Comment:
Move `keepBody` here, too:
```suggestion
            // Keep a full multi-line message body within the last six months (by default).
            // Note also that if body and subject are identical (single line), the body never need to be stored
           bool keepBody = authorUnixTime >= _oldestBody;
```